src/index.js

```diff
@@ -55,14 +62,18 @@ export default async (
               await cache.set(urlString, html);
             }
 
-            const preserve = ['position'];
-            for (const key of Object.keys(paragraphNode)) {
-              if (!preserve.includes(key)) {
-                delete paragraphNode[key];
-              }
-            }
-            paragraphNode.type = 'html';
-            paragraphNode.value = html;
+            // if nothing's returned from getHTML, then no modifications are needed
+            if (!html) return;
+
+            // convert the HTML string into an AST
+            const htmlElement = htmlToHast(html);
+
+            // set the paragraphNode.data with the necessary properties
```

Comment:
@wooorm, thanks so much for your help on this one. I'm not sure I understand how this works (I'm pretty sure the `paragaphNode`'s `type` is still `paragraph`, but somehow it changes to the `data.hName` property when the AST is compiled to HTML 🤷‍♂️

In any case, this is great! And it's nice to know that we're future-proof and completely backward compatible :)

Reply:
No problem!

The paragraph is till a paragraph. But it defined (though `.data`) that it should turn into something else when going to HTML. It defines an element name, props, and children for that!

So, when going from markdown to HTML, the process sees that there is something else defined rather than the normal markdown children, and it’ll use that instead